UI/Pages/Index.cshtml.cs

```diff
@@ -24,6 +24,16 @@ public Index(ILogger<Index> logger, IMediator mediator)
 
     public record Command : SearchModel, IRequest<Domain.IResult> { }
 
+    public IActionResult OnGet()
```

Comment:
This seems reasonable to me, we want the postcode to be optional for `GET` but mandatory for `POST`.  

I suppose we could do away with the `Data` property and take the postcode as a parameter:

    public void OnGet(string? postcode) {}
    public void OnPost(string postcode) {}

However we would then have to create the `Command` for mediator and it all sounds more complicated that simply suppressing the validation in `OnGet`.